python/taichi/_lib/utils.py

```diff
@@ -164,3 +171,77 @@ def _print_taichi_header():
 
 
 _print_taichi_header()
+
+
+def try_get_wheel_tag(module):
+    try:
+        import wheel.metadata  # pylint: disable=import-outside-toplevel
+        wheel_path = f'{module.__path__[0]}-{".".join(map(str, module.__version__))}.dist-info/WHEEL'
+        meta = wheel.metadata.read_pkg_info(wheel_path)
+        return meta.get('Tag')
+    except Exception:
+        return None
+
+
+def try_get_loaded_libc_version():
+    assert platform.system() == "Linux"
+    with open('/proc/self/maps') as f:
+        content = f.read()
+
+    try:
+        libc_path = next(v for v in content.split() if 'libc-' in v)
+        ver = re.findall(r'\d+\.\d+', libc_path)
+        if not ver:
+            return None
+        return tuple([int(v) for v in ver[0].split('.')])
+    except StopIteration:
+        return None
+
+
+def try_get_pip_version():
+    try:
+        import pip  # pylint: disable=import-outside-toplevel
+        return tuple([int(v) for v in pip.__version__.split('.')])
+    except ImportError:
+        return None
+
+
+def warn_restricted_version():
+    if os.environ.get('TI_MANYLINUX2014_OK', ''):
+        return
+
+    if get_os_name() == 'linux':
+        try:
+            import taichi as ti  # pylint: disable=import-outside-toplevel
+            wheel_tag = try_get_wheel_tag(ti)
+            if wheel_tag and 'manylinux2014' in wheel_tag:
+                print_yellow_bold(
+                    "You have installed a restricted version of taichi, certain features (e.g. Vulkan & GGUI) will not work."
+                )
+                libc_ver = try_get_loaded_libc_version()
+                if libc_ver and libc_ver < (2, 27):
+                    print_yellow_bold(
+                        "!! Taichi requires glibc >= 2.27 to run, please try upgrading your OS to a recent one (e.g. Ubuntu 18.04 or later) if possible."
+                    )
+
+                pip_ver = try_get_pip_version()
+                if pip_ver and pip_ver < (20, 3, 0):
+                    print_yellow_bold(
+                        f"!! Your pip (version {'.'.join(map(str, pip_ver))}) is outdated (20.3.0 or later required), "
+                        "try upgrading pip and install taichi again.")
+                    print()
+                    print_yellow_bold(
+                        "    $ python3 -m pip install --upgrade pip")
+                    print_yellow_bold(
+                        "    $ python3 -m pip install --force-reinstall taichi"
+                    )
+                    print()
+
+                print_yellow_bold(
+                    "You can suppress this warning by setting the environment variable TI_MANYLINUX2014_OK=1."
+                )
+        except Exception:
+            pass
+
+
+warn_restricted_version()
```

Comment:
nit: is it better to only show this when `ti.vulkan` is requested in `ti.init()`? I'd assume many users who wish to stay on the old OS don't want to see this everytime taichi starts ;)

Reply:
I'd like to drive users away from old operating systems haha, if this nagging helps it will be great ;)
For those who just don't care or can't, there is a suppressing switch `export TI_MANYLINUX2014_OK=1`

Reply:
ok sgtm!

Reply:
And, end users can use GGUI with cuda backend I think? I have removed the original messages and left a pointer to the startup message, if it don't show up it could be even more confusing.

Reply:
Yea you're definitely right! cuda with GGUI could potentially be more confusing to users. :D